cfn-resources/stream-connection/cmd/resource/resource.go

```diff
@@ -0,0 +1,226 @@
+	connectionName := currentModel.ConnectionName
+	streamConnectionReq := NewStreamConnectionReq(ctx, currentModel)
+	streamConnResp, apiResp, err := conn.StreamsApi.UpdateStreamConnection(ctx, *projectID, *instanceName, *connectionName, streamConnectionReq).Execute()
+	if err != nil {
+		return handleError(apiResp, constants.UPDATE, err)
+	}
+
+	resourceModel := GetStreamConnectionModel(streamConnResp, currentModel)
+
+	return handler.ProgressEvent{
+		OperationStatus: handler.Success,
+		Message:         "Update Completed",
+		ResourceModel:   resourceModel,
+	}, nil
+}
+
+func Delete(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
+	conn, peErr := setup(constants.DELETE, req, currentModel)
+	if peErr != nil {
+		return *peErr, nil
+	}
+
+	ctx := context.Background()
+
+	projectID := currentModel.ProjectId
+	instanceName := currentModel.InstanceName
+	connectionName := currentModel.ConnectionName
+	if _, apiResp, err := conn.StreamsApi.DeleteStreamConnection(ctx, *projectID, *instanceName, *connectionName).Execute(); err != nil {
+		return handleError(apiResp, constants.DELETE, err)
+	}
+
+	return handler.ProgressEvent{
+		OperationStatus: handler.Success,
+		Message:         "Delete Completed",
+		ResourceModel:   nil}, nil
+}
+
+func List(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
+	conn, peErr := setup(constants.LIST, req, currentModel)
+	if peErr != nil {
+		return *peErr, nil
+	}
+
+	ctx := context.Background()
+
+	projectID := currentModel.ProjectId
+	instanceName := currentModel.InstanceName
+
+	accumulatedStreamConns, apiResp, err := getAllStreamConnections(ctx, conn, *projectID, *instanceName)
+	if err != nil {
+		return handleError(apiResp, constants.LIST, err)
+	}
+
+	response := make([]interface{}, 0)
+	for i := range accumulatedStreamConns {
+		model := GetStreamConnectionModel(&accumulatedStreamConns[i], nil)
+		model.ProjectId = currentModel.ProjectId
+		model.InstanceName = currentModel.InstanceName
+		model.Profile = currentModel.Profile
+
+		response = append(response, model)
+	}
+
+	return handler.ProgressEvent{
+		OperationStatus: handler.Success,
+		ResourceModels:  response,
+	}, nil
+}
+
+func getAllStreamConnections(ctx context.Context, conn *admin.APIClient, projectID, instanceName string) ([]admin.StreamsConnection, *http.Response, error) {
+	pageNum := 1
+	accumulatedStreamConns := make([]admin.StreamsConnection, 0)
+	for ok := true; ok; {
+		streamConns, apiResp, err := conn.StreamsApi.ListStreamConnectionsWithParams(ctx, &admin.ListStreamConnectionsApiParams{
+			GroupId:      projectID,
+			TenantName:   instanceName,
+			ItemsPerPage: util.Pointer(100),
+			PageNum:      util.Pointer(pageNum),
+		}).Execute()
+
+		if err != nil {
+			return nil, apiResp, err
+		}
+		accumulatedStreamConns = append(accumulatedStreamConns, streamConns.GetResults()...)
+		ok = streamConns.GetTotalCount() > len(accumulatedStreamConns)
```

Comment:
I left a [similar question](https://github.com/mongodb/mongodbatlas-cloudformation-resources/pull/930) here for Oriol and I see it's the same logic, so I am probably missing something but I'd be curious to know how it works since the documentation of `totalCount` is not clear to me

Reply:
Yes we discussed this approach for List operation. Code has been updated as per comment on Oriol's PR, let me know if you have further questions. To summarize, `totalCount` will return number of records in all.